web/static/models/endpoint-model.js

```diff
@@ -0,0 +1,112 @@
+/**
+ * Enum for Globus endpoint entity types
+ * @readonly
+ * @enum {string}
+ */
+const EndpointEntityType = Object.freeze({
+  GCP_MAPPED_COLLECTION: 'GCP_mapped_collection',
+  GCP_GUEST_COLLECTION: 'GCP_guest_collection',
+  GCSV5_ENDPOINT: 'GCSv5_endpoint',
+  GCSV5_MAPPED_COLLECTION: 'GCSv5_mapped_collection',
+  GCSV5_GUEST_COLLECTION: 'GCSv5_guest_collection',
+  GCSV4_HOST: 'GCSv4_host',
+  GCSV4_SHARE: 'GCSv4_share',
```

Comment:
Probably worth including a link to the types:

https://docs.globus.org/api/transfer/endpoints_and_collections/

in the documentation.